libmavconn/src/serial.cpp

```diff
@@ -82,12 +90,16 @@ void MAVConnSerial::close()
 	if (!is_open())
 		return;
 
-	io_service.stop();
+	serial_dev.cancel();
 	serial_dev.close();
 
+	io_service.stop();
+
 	if (io_thread.joinable())
 		io_thread.join();
 
+	io_service.reset();
```

Comment:
That is not necessary, destructor should free automatically.

Reply:
Normally yes, but that way we ensure that it is really done ! Using serial on asio should also put the port on raw mode but isn't done properly ;-P see https://github.com/chriskohlhoff/asio/commit/ff9d69e2fc002f62828d1c23b5eaae993427bc07